python/lsst/fgcmcal/fgcmBuildStarsBase.py

```diff
@@ -554,22 +545,32 @@ def _fillVisitCatalog(self, visitCat, groupedDataRefs, bkgDataRefDict=None,
                 if visitCatDataRef is not None:
                     visitCatDataRef.put(visitCat)
 
-            # Note that the reference ccd is first in the list (if available).
-
-            # The first dataRef in the group will be the reference ccd (if available)
             dataRef = groupedDataRefs[visit][0]
             if isinstance(dataRef, dafPersist.ButlerDataRef):
+                # Gen2: calexp dataRef
+                # The first dataRef in the group will be the reference ccd (if available)
                 exp = dataRef.get(datasetType='calexp_sub', bbox=bbox)
                 visitInfo = exp.getInfo().getVisitInfo()
                 label = dataRef.get(datasetType='calexp_filterLabel')
                 physicalFilter = label.physicalLabel
                 psf = exp.getPsf()
+                psfSigma = psf.computeShape().getDeterminantRadius()
             else:
-                visitInfo = dataRef.get(component='visitInfo')
-                # TODO: When DM-28583 is fixed we can get the filterLabel
-                # via dataRef.get(component='filterLabel')
-                physicalFilter = dataRef.dataId['physical_filter']
-                psf = dataRef.get(component='psf')
+                # Gen3: use the visitSummary dataRef
+                summ = dataRef.get()
```

Comment:
Please don't call this "summ" - "visitSummary" makes it a lot clearer. I thought `summ['detector_id']` below was a typo on `sum()`.